maas/resource_maas_boot_resource.go

```diff
@@ -0,0 +1,330 @@
+package maas
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/canonical/gomaasclient/client"
+	"github.com/canonical/gomaasclient/entity"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ResourceMaasBootResources() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Provides a resource to manage MAAS bootresources.",
+		CreateContext: resourceBootResourcesCreate,
+		ReadContext:   resourceBootResourcesRead,
+		UpdateContext: resourceBootResourcesUpdate,
+		DeleteContext: resourceBootResourcesDelete,
+
+		Schema: map[string]*schema.Schema{
+			"boot_source": {
```

Comment:
Question: Since we want to hide the fact that we can have many boot_sources in the database, what is the benefit for exposing it here? How can this be useful for the Terraform practitioner?

Reply:
We use it equally for the selections, or are you suggesting we remove that entirely and just fetch the boot source each time, rather than reading from the terraform?